fog/view/server/src/fog_view_service.rs

```diff
@@ -151,6 +170,35 @@ impl<E: ViewEnclaveProxy, DB: RecoveryDb + Send + Sync> FogViewService<E, DB> {
         })
     }
 
+    fn process_queries(
+        &mut self,
+        fog_view_store_uri: FogViewStoreUri,
+        queries: Vec<attest::Message>,
+    ) -> MultiViewStoreQueryResponse {
+        let mut response = MultiViewStoreQueryResponse::new();
+        response.set_fog_view_store_uri(fog_view_store_uri.url().to_string());
+        for query in queries.into_iter() {
+            let result = self.query_impl(query);
+            if let Ok(attested_message) = result {
+                {
+                    let shared_state = self.db_poll_shared_state.lock().expect("mutex poisoned");
+                    if !self
+                        .sharding_strategy
+                        .is_ready_to_serve_tx_outs(shared_state.processed_block_count.into())
+                    {
+                        response.set_error(MultiViewStoreQueryResponseError::NOT_READY);
+                    } else {
+                        response.set_query_response(attested_message);
+                    }
+                }
+                return response;
```

Comment:
It was like this before so maybe I missed it or I'm just not recollecting the full behavior.  This seems to `return response` on the first response it's able to authenticate.  Is there something that prevents multiple serviceable queries coming to this shard?.  i.e. what if there were 2 queries this shard could handle it would return with the first one and not answer the second based on the code here.

Reply:
Yes this is the correct behavior. There should only be one response that the shard can decrypt and fulfill. There should not be any second response that it can decrypt.

Reply:
Could you add a comment to this effect somewhere (i.e. only one of the query messages in an MVSQ is intended for this store)?